src/cmd/commit.ts

```diff
@@ -0,0 +1,105 @@
+import { Argv } from 'yargs'
+import { $, cd } from 'zx'
+import { OtomiDebugger, terminal } from '../common/debug'
+import { env } from '../common/envalid'
+import { hfValues } from '../common/hf'
+import { capitalize, setParsedArgs } from '../common/no-deps'
+import { cleanupHandler, otomi, PrepareEnvironmentOptions } from '../common/setup'
+import { Arguments as HelmArgs, helmOptions } from '../common/yargs-opts'
+import { Arguments as DroneArgs, genDrone } from './gen-drone'
+import { validateValues } from './validate-values'
+
+const fileName = 'commit'
+let debug: OtomiDebugger
+
+interface Arguments extends HelmArgs, DroneArgs {}
+
+/* eslint-disable no-useless-return */
+const cleanup = (argv: Arguments): void => {
+  if (argv.skipCleanup) return
+}
+/* eslint-enable no-useless-return */
+
+const setup = async (argv: Arguments, options?: PrepareEnvironmentOptions): Promise<void> => {
+  if (argv._[0] === fileName) cleanupHandler(() => cleanup(argv))
+  debug = terminal(fileName)
+
+  if (options) await otomi.prepareEnvironment(options)
+  otomi.closeIfInCore(fileName)
+}
+
+export const preCommit = async (argv: DroneArgs): Promise<void> => {
+  const pcDebug = terminal('Pre Commit')
+  pcDebug.info('Check for cluster diffs')
+  const settingsDiff = (await $`git diff env/settings.yaml`).stdout.trim()
+  const secretDiff = (await $`git diff env/secrets.settings.yaml`).stdout.trim()
+
+  const versionChanges = settingsDiff.includes('+    version:')
+  const secretChanges = secretDiff.includes('+        url: https://hooks.slack.com/')
```

Comment:
If file is encrypted the you want see url: https://hooks.slack.com/'
Instead you will see diff like:
```    
        url: ENC[AES256_GCM,data: ...... ,type:str]
```

Reply:
Perhaps we could you a great tool called dyff that allows to discover changes in YAML file by providing JSONPath

E.g.:
```
dyff -c off between -o brief --set-exit-code --filter alerts.slack.url y1.yaml y2.yaml
```

Reply:
not an issues as sops diff plugin is used